### ISSUE TITLE
Rename "Mail" icon or add alternate name to cover "Email"

### ISSUE THREAD

**Issue by mrwweb** [WordPress/gutenberg#21733]
**Is your feature request related to a problem? Please describe.**
While not strictly "social media", email is often included in a row of social icons, especially on a staff/team page where this block is especially useful.

**Describe the solution you'd like**
An email icon (probably a postal letter or "@" symbol) that outputs a mailto: link would be great. Running that email through `antispambot()` would be even better!

**Comment by paaljoachim** [comment]
I'll ping Marcus @mkaz as he has been doing a lot of work with the Social Block.

**Comment by mkaz** [comment]
The icon is already there, its just called Mail and not Email.

![social-mailto](https://user-images.githubusercontent.com/45363/80138407-a1b81200-8559-11ea-8dd7-cc9fbc13c13a.gif)

**Comment by mrwweb** [comment]
@mkaz Thanks for the clarification. I've updated the title to cover my issue (not being able to find it).

**Comment by mkaz** [comment]
@jasmussen What do you think of adding an additional icon named "email" that is an "@" ? This would give someone the choice of type of email icons.

I also think a helper to prefill the `mailto:` portion would be nice, could be useful for many sites. For example Twitter could use a prefill: `https://twitter.com/`

**Comment by mrwweb** [comment]
re: Icon - FWIW, I don't care that much about a second icon. I only mentioned it since I thought one needed to be designed.

re: "mailto: - I didn't realize there _wasn't_ a prefill. The placeholder text says "enter email" so that's all I did. I think that is an absolute requirement.

And if we're adding one helper, I just want to plug one more time, if it's easy to obfuscate the address a la `antispambot()`, there's no excuse not to do it!

**Comment by jasmussen** [comment]
There are a number of ways to go about this. Aside from the direct path — adding an @ icon to the new Icon component first, then adding a new generic "@ address" child block for social links. That seems to be the task attached to this ticket.

An additional path is to enable this without needing to wait for this ticket to be addressed, then for the PR to land in the plugin and ultimately WordPress. Specifically:

- The Icon component is already extensible. This should be possible using `<Icon icon={ <svg mysvg... /> } />` — the Material icon for @ slots right in: https://material.io/resources/icons/?search=@&icon=alternate_email&style=baseline
- Social Links should be extensible per #19041 

Combining the above two the goal would be to enable anyone to add a social link block for any purpose, or even for a plugin to extend the social links block.

**Comment by mrwweb** [comment]
@jasmussen @mkaz I'd like to refocus this ticket on my primary concern and the reason I opened it.

**Searching the social icons for "email" does not have a result.** This lead me to think that email wasn't an option.

Here's the issue summarized in a screenshot:

![social icon block search for "email" provides no results](https://user-images.githubusercontent.com/1238696/80233085-cddf9b80-860a-11ea-8730-49c64c75e777.png)

I tried renaming the ticket to make this clearer, but feel free to re-rename to whatever will clarify the purpose of the ticket.

As @jasmussen mentions, there's already a ticket for extensibility, and I will open up a separate ticket for "mailto:" and obfuscation.
